server/src/main/java/org/opensearch/action/search/TransportDeletePitAction.java

```diff
@@ -86,16 +85,12 @@ private void deletePits(ActionListener<DeletePitResponse> listener, DeletePitReq
      * Delete all active PIT reader contexts
      */
     private void deleteAllPits(ActionListener<DeletePitResponse> listener) {
-        // TODO: Use list all PITs to delete all PITs in case of remote cluster use case
-        int size = clusterService.state().getNodes().getSize();
-        ActionListener groupedActionListener = pitService.getDeletePitGroupedListener(listener, size);
-        for (final DiscoveryNode node : clusterService.state().getNodes()) {
-            try {
-                Transport.Connection connection = searchTransportService.getConnection(null, node);
-                searchTransportService.sendFreeAllPitContexts(connection, groupedActionListener);
-            } catch (Exception e) {
-                groupedActionListener.onFailure(e);
-            }
-        }
+        // Get all PITs and execute delete operation for the PITs
+        pitService.getAllPits(ActionListener.wrap(getAllPitNodesResponse -> {
+            DeletePitRequest deletePitRequest = new DeletePitRequest(
+                getAllPitNodesResponse.getPITIDs().stream().map(r -> r.getPitId()).collect(Collectors.toList())
```

Comment:
nit: Lets follow standard consistent naming `PIT` vs `Pit`

Reply:
WIth this change will increase the N/W round trip to first get all and then do delete all

Reply:
Yes correct - also as discussed, with this change mixed cluster PITs will be fully deleted.
As previous logic deletes contexts only in local cluster.
* Still fully remote PITs will not be deleted * with 'delete all' as 'list all pits' retrieves PIT IDs from current cluster only.

That's the tradeoff.

Reply:
@bharath-techie I think there is disconnect between what you are stating and implementation, in particular looking into `PitService::deletePitContexts`: when the delete operation fails, it seems like we still report it as success, enumerating PITs which may have not been deleted:

```
                   try {
                        final Transport.Connection connection = searchTransportService.getConnection(clusterAlias, node);
                        searchTransportService.sendFreePITContexts(connection, entry.getValue(), groupedListener);
                    } catch (Exception e) {
                        logger.error(() -> new ParameterizedMessage("Delete PITs failed on node [{}]", node.getName()), e);
                        List<DeletePitInfo> deletePitInfos = new ArrayList<>();
                        for (PitSearchContextIdForNode pitSearchContextIdForNode : entry.getValue()) {
                            deletePitInfos.add(new DeletePitInfo(false, pitSearchContextIdForNode.getPitId()));
                        }
                        groupedListener.onResponse(new DeletePitResponse(deletePitInfos));
                    }
```

It looks to me that the implementation is not doing what tit should be: report an error for such PITs

Reply:
Delete PIT by ID will completely delete the PIT context. Even fully remote ones - since ID contains node mapping.

Delete all meanwhile will delete local and mixed PITs - as list all pits doesn't fetch pits from remote clusters - which is leveraged by delete all.

Reply:
` deletePitInfos.add(new DeletePitInfo(false, pitSearchContextIdForNode.getPitId()));`

Also, for failures, we enumerate them by initializing 'DeletePitInfo' with 'successful' as false - the flag for each PIT which lets the user know that there are failures in deletion.

Reply:
Oh, right `new DeletePitInfo(false, ..)`, :+1: